src/Views/MainView.vala

```diff
@@ -195,4 +257,39 @@ public class SwitchboardPlugUserAccounts.Widgets.MainView : Gtk.Box {
             content.set_visible_child_name ("guest_session");
         }
     }
+
+    private void update_listbox () {
+        foreach (unowned var useritem in listbox.get_children ()) {
+            listbox.remove (useritem);
+        }
+
+        listbox.insert (new UserItem (get_current_user ()), 0);
+        int pos = 1;
+        foreach (unowned Act.User temp_user in get_usermanager ().list_users ()) {
+            if (get_current_user () != temp_user && !check_removal (temp_user)) {
+                listbox.insert (new UserItem (temp_user), pos);
+                pos++;
+            }
+        }
+
+        listbox.insert (guest_session_row, pos);
```

Comment:
Should check that `guest_session_row` exists?  Alternatively insert a dummy row indicating that a guest session is unavailable.